src/dependencies.jl

```diff
@@ -263,7 +263,7 @@ provider(::Type{Binaries},uri::URI; opts...) = RemoteBinaries(uri)
 provider(::Type{Binaries},path::AbstractString; opts...) = CustomPathBinaries(path)
 provider(::Type{SimpleBuild},steps; opts...) = SimpleBuild(steps)
 provider{T<:BuildProcess}(::Type{BuildProcess},p::T; opts...) = provider(T,p; opts...)
-provider(::Type{BuildProcess},steps::Union(BuildStep,SynchronousStepCollection); opts...) = provider(SimpleBuild,steps; opts...)
+@compat provider(::Type{BuildProcess},steps::Union{BuildStep,SynchronousStepCollection}; opts...) = provider(SimpleBuild,steps; opts...)
```

Comment:
You mean the version requirement for Compat? The previous PR requires the latest Compat (0.7.1) so I'm pretty sure it's in.